### PR TITLE
feat: enable windows and macos build

### LINKED ISSUE #192: Add MacOS Support
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/96, add MacOS (Darwin) support.

- [ ] Passing CI on MacOS
- [ ] Documentation

### PR DESCRIPTION
**Pull request by derek-ho**
Signed-off-by: Derek Ho <dxho@amazon.com>

### Description
Enable windows and macos builds

### Issues Resolved
#192 
#191 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/reports-scheduler-test-and-build-workflow.yml [hunk 1]
```diff
@@ -47,5 +47,85 @@ jobs:
...
+
+      # reports-scheduler
+      - name: Checkout Reports Scheduler
+        uses: actions/checkout@v2
+
+      - name: RunBackwards Compatibility Tests
+        run: |
+          cd reports-scheduler
+          echo "Running backwards compatibility tests ..."
+          ./gradlew.bat bwcTestSuite
```
Comment (derek-ho) [inline .github/workflows/reports-scheduler-test-and-build-workflow.yml:75]: Getting the following error, is this available? : 
`Could not HEAD 'https://artifacts.opensearch.org/releases/core/opensearch/1.1.0/opensearch-min-1.1.0-windows-x64.zip'. Received status code 403 from server: Forbidden`

Comment (derek-ho) [reply .github/workflows/reports-scheduler-test-and-build-workflow.yml:75]: Same thing is happening for windows and macos

Comment (joshuali925) [reply .github/workflows/reports-scheduler-test-and-build-workflow.yml:75]: see https://github.com/opensearch-project/observability/pull/1108#discussion_r993856324

File: .github/workflows/dashboards-reports-test-and-build-workflow.yml [hunk 2]
```diff
@@ -111,8 +103,148 @@ jobs:
...
+          curl.exe -LO https://github.com/opendistro-for-elasticsearch/kibana-reports/releases/download/chromium-1.12.0.0/chromium-windows-x64.zip
+          Expand-Archive -Path 'chromium-windows-x64.zip'
+          rd chromium-windows-x64.zip
+
+      - name: OpenSearch Dashboards Plugin Bootstrap
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 30
+          max_attempts: 3
+          command: cd OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}; yarn osd bootstrap
```
Comment (joshuali925) [inline .github/workflows/dashboards-reports-test-and-build-workflow.yml:150]:
```suggestion
          command: cd OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }} && yarn osd bootstrap
```

File: .github/workflows/dashboards-reports-test-and-build-workflow.yml [hunk 3]
```diff
@@ -111,8 +103,148 @@ jobs:
...
+
+      - name: Test
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 30
+          max_attempts: 3
+          command: cd OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}; yarn test
```
Comment (joshuali925) [inline .github/workflows/dashboards-reports-test-and-build-workflow.yml:157]: same here
```suggestion
          command: cd OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }} && yarn test
```

Comment (derek-ho) [reply .github/workflows/dashboards-reports-test-and-build-workflow.yml]: Thanks this worked

File: .github/workflows/dashboards-reports-test-and-build-workflow.yml (hunk 2 above)
Comment (derek-ho) [reply .github/workflows/dashboards-reports-test-and-build-workflow.yml]: Thanks this worked

**Comment by derek-ho** [comment]
This is blocked on incompatibility between windows chromium binary and puppeteer. CC @joshuali925

File: dashboards-reports/test/jest.config.js [hunk 4]
```diff
@@ -3,7 +3,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-process.env.TZ = 'America/Los_Angeles';
+//@ts-check
```
Comment (joshuali925) [inline dashboards-reports/test/jest.config.js:6]: this is irrelevant but i think it's ok

**Review by joshuali925 — APPROVED** [review]

**Review by rupal-bq — APPROVED** [review]

**Review by anirudha — APPROVED** [review]

**Pull request merged**